xamarin/Xamarin.Forms
# [GTK] Fix button shrinking

### Description of Change ###

Using buttons on page without navbar causes them to shrink to 1px width on GTK. Two things leading to this:

If a button desired size is bigger than available space, button [would shrink to width constraint](https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Platform.GTK/Extensions/WidgetExtensions.cs#L39) and would stay this way on next cycles, even if it could get more space;

Each page starts to allocate children sizes with its initial size 1px X 1px, on next cycles getting real window size. Pages with navbar would subtract navbar height from available size and get negative number, so they won't ask children to allocate space on this cycle. Pages without any navbar would calculate its children sizes on page 1px wide, stucking buttons with width=1px for all next cycles.

Overriding ButtonRendered.GetDesiredSize method to remove button size mutation fixed this behavior.

### Issues Resolved ### 

- fixes #9092 (this is the real reason behind the bug, not disabling button or changing navigation pages)

### API Changes ###

 None

### Platforms Affected ### 

- GTK

### Behavioral/Visual Changes ###

Buttons should stop to shrink

### Before/After Screenshots ### 

No navbar, green row on the top of the page

Before:
![Xamarin Forms GTK# Backend 13-Jan-20 22_19_32](https://user-images.githubusercontent.com/24920435/72338451-8d03ec80-36d5-11ea-9fd0-febe256d2705.png)

After:
![Xamarin Forms GTK# Backend 13-Jan-20 22_20_13](https://user-images.githubusercontent.com/24920435/72338463-91300a00-36d5-11ea-8c20-811a25af9276.png)

### Testing Procedure ###

[GTK] Add button on page without navbars. Confirm that button has right width. Change button text and WidthRequest to check size changing and constraint fitting.

Tested this behavior on Windows 10 and Kubuntu 19.04

### PR Checklist ###
